src/theme-mode-script.tsx

```diff
@@ -0,0 +1,42 @@
+import React from 'react';
+import type { ThemeMode } from './helpers/use-theme-mode';
+
+export interface ThemeModeScriptProps extends React.ComponentPropsWithoutRef<'script'> {
+  mode?: ThemeMode;
+}
+
+export const ThemeModeScript = ({ mode, ...others }: ThemeModeScriptProps) => {
+  return (
+    <script
+      {...others}
+      data-flowbite-theme-mode-script
+      dangerouslySetInnerHTML={{
+        __html: getScript({ mode, defaultMode: 'light', localStorageKey: 'flowbite-theme-mode' }),
+      }}
+    />
+  );
+};
+
+function getScript({
+  mode,
+  defaultMode,
+  localStorageKey,
+}: {
+  mode?: ThemeMode;
+  defaultMode: ThemeMode;
+  localStorageKey: string;
+}) {
+  return `
+    try {
+      const mode = window.localStorage.getItem('${localStorageKey}') ?? '${mode}' ?? '${defaultMode}';
+      const computedMode =
+        mode === 'auto' ? (window.matchMedia('(prefers-color-scheme: dark)').matches ? 'dark' : 'light') : mode;
+
+      if (computedMode === 'dark') {
+        document.documentElement.classList.add('dark');
+      } else {
+        document.documentElement.classList.remove('dark');
+      }
+    } catch (e) {}
```

Comment:
Maybe we should log this? It could be helpful for users to report the error if anything happens

Reply:
The try catch block is solely here for the SSR side, in case it tries to execute this on server, it doesn't work.